File: frontend/src/types/court.ts
/** A 2D point with integer (original video space) or float (display space) coords. */
export interface Point {
  x: number
  y: number
}

/**
 * Court corners in order TL, TR, BR, BL — the order produced by CourtKeyNet
 * and consumed by the Homography/CourtMapper pipeline.
 */
export type Corner = [number, number]
export type Corners = Corner[]

export interface CourtDetectionResponse {
  success: boolean
  detector: string
  corners: Corners | null
  roi_corners?: Corners | null
  mid_height?: number | null
  preview_url?: string | null
  message?: string | null
}

export interface CourtState {
  corners: Corners | null
  detector: string
  roi_corners?: Corners | null
  mid_height?: number | null
  best_frame_path?: string | null
  preview_url?: string | null
}

export const CORNER_LABELS = ["TL", "TR", "BR", "BL"] as const

File: backend/schemas/court.py
# -*- coding: utf-8 -*-
from pydantic import BaseModel, Field
from typing import Optional


class CourtCorners(BaseModel):
    """Four corner points in ORIGINAL video pixel coordinates: TL, TR, BR, BL."""

    corners: list[list[int]] = Field(..., min_length=4, max_length=4)


class CourtState(BaseModel):
    corners: Optional[list[list[int]]] = None
    detector: str = "courtkeynet"
    roi_corners: Optional[list[list[int]]] = None
    mid_height: Optional[float] = None
    best_frame_path: Optional[str] = None
    preview_url: Optional[str] = None
